packages/ember-glimmer/lib/syntax/mount.js

```diff
@@ -0,0 +1,101 @@
+import { ArgsSyntax, StatementSyntax } from 'glimmer-runtime';
+import { ConstReference, isConst } from 'glimmer-reference';
+import { assert } from 'ember-metal/debug';
+import { RootReference } from '../utils/references';
+import { generateControllerFactory } from 'ember-routing/system/generate_controller';
+import { OutletLayoutCompiler } from './outlet';
+
+function makeComponentDefinition(vm) {
+  let env     = vm.env;
+  let args    = vm.getArgs();
+  let nameRef = args.positional.at(0);
+
+  assert(`The first argument of {{mount}} must be quoted, e.g. {{mount "chat-engine"}}.`, isConst(nameRef));
+
+  let engineName = nameRef.value();
+
+  assert(`You used \`{{mount '${engineName}'}}\`, but '${engineName}' can not be found as an engine.`, env.owner.hasRegistration(`engine:${engineName}`));
+
+  let engine = env.owner.lookup(`engine:${engineName}`);
+
+  return new ConstReference(new MountDefinition(engineName, engine, env, MOUNT_MANAGER));
+}
+
+export class MountSyntax extends StatementSyntax {
+  constructor({ args, symbolTable }) {
+    super();
+    this.definitionArgs = args;
+    this.definition = makeComponentDefinition;
+    this.args = ArgsSyntax.fromPositionalArgs(args.positional.slice(1, 2));
+    this.templates = null;
+    this.symbolTable = symbolTable;
+    this.shadow = null;
+  }
+
+  compile(builder) {
+    builder.component.dynamic(this.definitionArgs, this.definition, this.args, this.templates, this.symbolTable, this.shadow);
+  }
+}
+
+class MountManager {
+  prepareArgs(definition, args) {
+    return args;
+  }
+
+  /* abstract create(definition, args, dynamicScope); */
+
+  layoutFor(definition, bucket, env) {
+    return env.getCompiledBlock(OutletLayoutCompiler, definition.template);
+  }
+
+  getSelf({ controller }) {
+    return new RootReference(controller);
+  }
+
+  getTag(state) {
+    return null;
+  }
+
+  getDestructor(state) {
+    return state.controller;
+  }
+
+  didCreateElement() {}
+  didRenderLayout() {}
+  didCreate(state) {}
+  update(state, args, dynamicScope) {}
+  didUpdateLayout() {}
+  didUpdate(state) {}
+
+  create(definition, args, dynamicScope) {
+    let { name, env } = definition;
+    let modelRef = args.positional.at(0);
+
+    let factory = env.owner._lookupFactory(`controller:${name}`) || generateControllerFactory(env.owner, name);
+    let controller = factory.create({ model: modelRef.value() });
+
+    if (dynamicScope.rootOutletState) {
+      dynamicScope.outletState = dynamicScope.rootOutletState.getOrphan(name);
```

Comment:
I am unsure if we need this. Seems like we generally do not support {{outlet inside of a {{mount....